modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/internal/TaskManager.kt

```diff
@@ -79,7 +79,17 @@ class TaskManager @Inject constructor(val args: Args,
     }
 
     fun runTargets(passedTaskNames: List<String>, allProjects: List<Project>): RunTargetResult {
-        val taskInfos = calculateDependentTaskNames(passedTaskNames, allProjects)
+        // Check whether tasks passed at command line exist
+        passedTaskNames.forEach {
+            if (!hasTask(TaskInfo(it)))
+                throw KobaltException("Unknown task: $it")
```

Comment:
The check in `runTarget()` checks the tasks passed at command line **before**
```
taskInfos = taskInfos.filter { hasTask(it) }
```
removes non existing tasks.
Without this check, invalid tasks passed at command line would never be reported.